src/renderer/hooks/useMapServerQuery.js

```diff
@@ -0,0 +1,22 @@
+// @ts-check
+import ky from 'ky/umd'
+import { useQuery } from '@tanstack/react-query'
+
+// local host
+// global port number
+const MAP_SERVER_URL = 'http://127.0.0.1:' + window.mapServerPort
+
+/**
+ * @param {'/styles' | `/styles/${string}` | '/tilesets' | `/tilesets/${string}`} resourcePath URL path to resource on Map Server (needs to start with `/`)
+ */
+export function useMapServerQuery (resourcePath) {
+  return useQuery({
+    queryKey: [resourcePath],
+    queryFn: () => ky.get(MAP_SERVER_URL + resourcePath).json()
+  })
+}
+
+/**
+ * @type {import('@tanstack/react-query').UseQueryResult<ReturnType<import('@mapeo/map-server/dist/api/styles').StylesApi["listStyles"]>>}
+ */
+const { data } = useMapServerQuery('/styles')
```

Comment:
if we don't create an API layer, we will need to type every `useMapServerQuery` @gmaclennan

Reply:
I think it's best not to add more code just to solve a typing issue.

I think the easiest way to do this is with function overloads ([example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAawJ5wGZQiOByZ2BQokscA3nAK4DOApgIoU1QoC+aGW2AAjAIYB2VPgGMEAeig1ewmAFoAjo2YF8+VBX4zgEfpVoBZXmADKTAG5MGTJAApJVCBSjCaABV4wAFgC4cYoUgANjRU2ACUvgCCUFC8SAA85MAAJr5CUMD8AOZwLAB8ahpaOno0hibmlkq29o7Obh4+cAAG-jBBIWIAJKTpmVkszRFkcPy8IDRpMBnZuYWaMNq61GVGplAWUFbMdiF1Lu5eUzNZwwIopPhwcJIwTsu027aX19eK1gDSNEi+ANq1TgOjQAugAaK6vd7MABi-F8NjCcAAvHlEEgAHRZGgwGzYTwwGBgbxiMSBCDCXiBTwQITeAAcAAZGdg4ABqG57QENLxhdEAKwc-AREJYYXwLFUwh0QjgAWCVAAMsAZUjSuV1psnri2h1QmKpYJ4HKaABJfioCDItVrSpbara41UMS8ABGwnC+ANMsyZkpKStK3Vtq12DEPr9yUOnnCcBJcHaYBCwgyYHgTAwUEl0ogwXRZKyNkdSqEvwZwPRKURcf4lvT0FB8aQibgytl036nuzufzhfawTNFvRYwmVbEo0tCaTKbTMXrjebrb62XwQA)).

I don't think this can be done in JSDoc though, so would require a typescript file to define the type and then import here.

Reply:
I understand how to do this typescript, but I am not sure I fully understand how we would implement this is JSDOC. I would create a `ts` file define everything there, import it into a `js` file, then export it, almost like an alias? Is that what you are suggesting?